astropy/table/row.py

```diff
@@ -38,10 +38,24 @@ def __init__(self, table, index):
                              .format(index, len(table)))
 
     def __getitem__(self, item):
-        return self._table.columns[item][self._index]
+        if self._table._is_list_or_tuple_of_colnames(item):
+            cols = [self._table[name] for name in item]
+            out = self._table.__class__(cols, copy=False)[self._index]
+
+        else:
+            out = self._table.columns[item][self._index]
+
+        return out
 
     def __setitem__(self, item, val):
-        self._table.columns[item][self._index] = val
+        if self._table._is_list_or_tuple_of_colnames(item):
+            if len(val) != len(item):
```

Comment:
I'm not sure I see the point of this check: if someone puts in a scalar, they now get a error that is not all that obvious, and it excludes iterables without a length. Maybe best to just over the relevant part of `Table.__setitem__` so things are consistent. Alternatively, maybe we pass this back to `Table`, by just doing
```
self._table[item][self._index] = val
```
But that perhaps has more overhead than warranted.

Reply:
Interesting point which has a lot of implications:
- I had not considered the idea of essentially broadcasting a scalar.  This is not supported in `insert_row`, and it seems like an unusual thing to do.  Agreed that catching this more nicely would be good.
- Likewise the iterable for row values seems obscure.  It is certainly important that the supplied values exactly match in length the number of columns.  I guess iterables could be supported by checking for an iterable and doing `tuple(..)` around it?
- The goal (admittedly not entirely achieved) is to make setting the row transactional, so if it fails half way you don't half-way update the table.  `insert_row` does do this.  The existing code here catches two common errors up front, namely bad column name and mismatch between columns and values.  What it doesn't catch is where setting fails (`self._table.columns[name][self._index] = val_`).  So that should be fixed.

Anyway, point taken, I'm thinking about it.

Reply:
How about this for the first two bullets, which basically says (nicely) "don't do that".  I think it is not an unreasonable restriction that the right side is a sequence (with a length) and that it match.
```
            try:
                assert len(val) == len(item)
            except Exception:
                raise ValueError('right hand side must be a sequence of values with '
                                 'the same length as the number of selected columns')
```

Reply:
Yes, this does the same as what you have in table but is cleaner. Good enough for now!